op-deployer/pkg/deployer/apply.go

```diff
@@ -58,6 +58,12 @@ func (a *ApplyConfig) Check() error {
 		return fmt.Errorf("logger must be specified")
 	}
 
+	if a.DeploymentTarget == DeploymentTargetGenesis {
+		if a.L1RPCUrl != "" {
+			return fmt.Errorf("l1-rpc-url should not be specified when deployment-target is genesis")
```

Comment:
https://github.com/ethereum-optimism/optimism/blob/25157554bbe0ff8a5d8b3691e4be2f70e686673a/op-deployer/pkg/deployer/flags.go#L60-L65

1. Does this change mean that we need to change from "can be blank" to "must be blank" here?

2. What should the behaviour be for the other deployment targets?

3. Why should we get an error if we supply something that isn't needed? Couldn't we just log a warning and continue?